mesonbuild/mintro.py

```diff
@@ -276,9 +271,6 @@ def list_target_files(target_name, targets, builddata: build.Build):
     for i in tgt['target_sources']:
         result += i['sources'] + i['generated_sources']
 
-    # TODO Remove this line in a future PR with other breaking changes
-    result = list(map(lambda x: os.path.relpath(x, builddata.environment.get_source_dir()), result))
```

Comment:
For my point of view `--target-files` is no longer really needed. So i would just keep that frozen. It does not seem that this line would be much of a performance or maintenance burden.

Reply:
I know, but it is a bit annoying that we have to load `builddata` just for this function.

> For my point of view `--target-files` is no longer really needed.

Maybe we can deprecate/remove it then? I did a [GitHub search](https://github.com/search?q=--target-files&type=Code) for `--target-files` and no one seems to be using it. That doesn't mean that no one elsewhere is using it, but I think that deprecating ore even removing it would be safe.

Reply:
I mean, no one seems to be using it and we are breaking everyone's introspection implementation anyway. So why not go all out and remove it completely?

Now that I think about it, this API cannot be used without knowing the target ID, which can only be obtained with the `--targets` API that already includes the information of `--target-files`.

It is also the only API that is not saved to disk, and that takes a command line argument. This makes it to the one od ball function in #4732 that requires special treatment.

Reply:
I hope this does not break all users of introspection. If it does we have to think hard about alternatives.

We have to check if gnome-builder uses this. But grepping some binary it seems it does not contain this string...

Reply:
OK. Done with the last commit.